docs/flow-results-connector.md

```diff
@@ -0,0 +1,112 @@
+
+```
+{
+  "form_uuid": "[package form UUID]",
+  "table_name": "[preferred Postgres table name]",
+  "base_url": "[Base URL to get data from]"	
+}
+```
+
+
+> Note on API authentication
+> For APIs that require authentication, edit processors that make GET requests to the API for survey and response named `GET flow results survey from API` and `GET flow results responses from API`. Enter the Basic Authentication Username and Basic Authentication Password or add a property for Authorization token by clicking + in properties.
+
+
+
+2. __Survey and response JSON feed__
+For JSON feed, under `Flow results JSON` processor, add the survey JSON to survey attribute and response JSON to the response attribute and custom text property.
+
+
+
+3. __Pushed submissions__
+To push a FLOIP submission, a POST request is made to NiFi's URL port 9090 and `floip` endpoint. The package name or initially defined table name should be specified as part of the request headers which determines the Postgres table to be updated. Below is an example of a curl request:
+
+
+```
+curl http://localhost:9090/floip -H "package_name: pizza_survey" -d '[["2019-07-09T13:47:05+00:00",1171290623824092,"john","cb398d5a-a039-4ee1-9ffb-16c6c68a8b4d","date","2019-07-04",null],["2019-07-09T13:47:05+00:00",1171290672224313,"john","cb398d5a-a039-4ee1-9ffb-16c6c68a8b4d","name","Pizzarea",null],["2019-07-09T13:47:05+00:00",1171290817424982,"john","cb398d5a-a039-4ee1-9ffb-16c6c68a8b4d","stars",4,null],["2019-07-09T13:47:05+00:00",1171291011025888,"john","cb398d5a-a039-4ee1-9ffb-16c6c68a8b4d","borough","Manhattan",null],["2019-07-09T13:47:05+00:00",1171291059426117,"john","cb398d5a-a039-4ee1-9ffb-16c6c68a8b4d","visited","yes",null],["2019-07-09T13:47:05+00:00",1171291156226578,"john","cb398d5a-a039-4ee1-9ffb-16c6c68a8b4d","location","-1.291449 36.787959 1 1",null]]'
+```
+
+
+
+## Python Modules
+The flow uses four scripts to flatten and transform flow results schema and responses:
+
+
+1. __Create table statement parser__
+This script takes in flow-results formatted survey together with table name if defined as parameters and returns a valid Postgres create table statement. The main function iterates through the `questions` object, getting column names from question key and column type from the type value. The column types are substituted from a dictionary containing a map of schema types to valid Postgres types. In the case of the table name not defined, the package attribute name is used. Below is an example of pizza survey schema:
+
+<img src="https://github.com/onaio/floip-canopy/blob/documentation/docs/images/pizza_survey_schema.png" alt="FLOIP flowchart" width="350" height="350">
```

Comment:
I would recommend we use the package descriptor JSON instead of a screenshot of it which is barely readable.